--- /usr/local/share/perl/5.8.4/Time/Out.pm	2007-09-14 15:39:31.000000000 -0400
+++ /usr/local/share/perl/5.8.4/Time/Out2.pm	2007-12-24 10:52:00.078207696 -0500
@@ -17,12 +17,12 @@
 $Time::Out::VERSION = '0.10' ;
 
 
-sub timeout($@){
-	my $secs = shift ;
+sub timeout(&$;$) {
+	my $secs = pop;
 	carp("Timeout value evaluates to 0: no timeout will be set") if ! $secs ;
-	my $code = pop ;
+	my $code = shift;
 	usage() unless ((defined($code))&&(UNIVERSAL::isa($code, 'CODE'))) ;
-	my @other_args = @_ ;
+	my @other_args = @{$_[0]} if @_;
 
 	my $prev_alarm = 0 ;
 	my $prev_time = 0 ;
