--- perl-5.10.0/t/op/smartmatch.t	2007-12-18 05:47:08.000000000 -0500
+++ smartmatch_.t	2009-12-19 20:52:26.000000000 -0500
@@ -1,8 +1,8 @@
 #!./perl
 
 BEGIN {
-    chdir 't';
-    @INC = '../lib';
+#    chdir 't';
+#    @INC = '../lib';
     require './test.pl';
 }
 use strict;
@@ -10,6 +10,8 @@
 use Tie::Array;
 use Tie::Hash;
 
+eval "use Match::Smart 'smart_match'" unless $] >= 5.009;
+
 # The feature mechanism is tested in t/lib/feature/smartmatch:
 # This file tests the semantics of the operator, without worrying
 # about feature issues such as scoping etc.
@@ -45,10 +47,12 @@
     die "Bad test spec: ($yn, $left, $right)"
 	unless $yn eq "" || $yn eq "!";
     
-    my $tstr = "$left ~~ $right";
+    #Bor-fackwards compatability
+    my $tstr = $] >= 5.009 ? "$left ~~ $right" : "smart_match($left, $right)";
     
     my $res;
-    $res = eval $tstr // "";	#/ <- fix syntax colouring
+    $res = eval $tstr || "";	#/ <- fix syntax colouring
+                      #         No defined-or joy for us in Obsolete-ville
 
     die $@ if $@ ne "";
     ok( ($yn =~ /!/ xor $res), "$tstr: $res");
